### DISCUSSION TITLE
Unity repository detected as ShaderLab instead of C#

### DISCUSSION THREAD

**Discussion by wapcrazut** [github-linguist/linguist#6682] [Classification] [answered]
Hello, I'm having a problem similar to this old issue: https://github.com/github-linguist/linguist/issues/5117

My Unity repository is detected as ShaderLab because there is an `Assets/TextMesh Pro/` directory with some code in it. 

Since that directory is created by a dependency I wanted to ignore it using the `.gitatributtes` config of [my repository](https://github.com/wapcrazut/unity-csharp-techniques/blob/master/.gitattributes)
```
# linguist overrides
Assets/TextMesh?Pro/* linguist-vendored
```

However, the configuration from above does not seem to be working, because the repository is still detected as ShaderLab I also waited a few days with other repositories but the issue persists.

Am I missing something?

Thank you in advance.

**Comment by lildude** [comment, marked as answer]
Your current override isn't sufficient. It's only covering the files directly in that directory. You need to use `/**` to recurse into the sub-directories.

From the [`.gitattributes` doc](https://git-scm.com/docs/gitattributes):

> The rules by which the pattern matches paths are the same as in .gitignore files (see [gitignore[5]](https://git-scm.com/docs/gitignore)), with a few exceptions:
>  - negative patterns are forbidden
>  - patterns that match a directory do not recursively match paths inside that directory (so using the trailing-slash `path/` syntax is pointless in an attributes file; use `path/**` instead)

From the [`.gitignore` doc](https://git-scm.com/docs/gitignore):

> An asterisk "`*`" matches anything except a slash. The character "`?`" matches any one character except "`/`". The range notation, e.g. `[a-zA-Z]`, can be used to match one of the characters in a range. See fnmatch(3) and the FNM_PATHNAME flag for a more detailed description.

... and ...

> A trailing "`/**`" matches everything inside. For example, "`abc/**`" matches all files inside directory "`abc`", relative to the location of the `.gitignore` file, with infinite depth.

**Comment by wapcrazut** [comment]
I was doubting that single * but didn't try to change it.
It worked, thank you for all the references @lildude 🏅
